server/src/main/java/org/candlepin/pinsetter/tasks/HypervisorUpdateJob.java

```diff
@@ -251,31 +268,58 @@ public void toExecute(JobExecutionContext context) throws JobExecutionException
                         hypervisorId.equalsIgnoreCase(knownHost.getHypervisorId().getHypervisorId()) &&
                         knownHost.getHypervisorId().getReporterId() != null &&
                         !jobReporterId.equalsIgnoreCase(knownHost.getHypervisorId().getReporterId())) {
+
                         log.debug("Reporter changed for Hypervisor {} of Owner {} from {} to {}",
                             hypervisorId, ownerKey, knownHost.getHypervisorId().getReporterId(),
                             jobReporterId);
                     }
-                    if (consumerResource.performConsumerUpdates(incoming, knownHost, guestConsumersMap,
-                        false)) {
-                        consumerCurator.update(knownHost);
-                        result.updated(updateCheckinTime(knownHost));
+
+                    boolean changesMade = consumerResource.checkForFactsUpdate(knownHost, incoming);
+                    changesMade = consumerResource.checkForGuestsUpdate(knownHost, incoming) || changesMade;
```

Comment:
We should add a TODO here to denote that we need to refactor this a bit so we're not calling into a resource for common work.

Also, slight aside, we're starting to diverge pretty heavily from what master has/does in this regard. That'll likely be a problem going forward.